ios/push-notifications.md

```diff
@@ -67,50 +60,107 @@ Use the following steps to connect add push notification backend services to you
-	          AWSMobileClient.default().initialize { (userState, error) in
-              if let error = error {
-                print("Error initializing AWSMobileClient: \(error.localizedDescription)")
-              } else if let userState = userState {
-                print("AWSMobileClient initialized. Current UserState: \(userState.rawValue)")
-              }
+            DispatchQueue.main.async {
+                // Register with Apple Push Notification service
+                UIApplication.shared.registerForRemoteNotifications()
             }
+        }
+    }
+}
+```
 
-            // Initialize Pinpoint
-            let pinpointConfiguration = AWSPinpointConfiguration.defaultPinpointConfiguration(launchOptions: launchOptions)
-            pinpoint = AWSPinpoint(configuration: pinpointConfiguration)
-            /** end code copy **/
-            return true
-       }
+2. Add the AppDelegate lifecycle methods to register the device token with AWS pinpoint, and handle push notifications from the foreground and background.
+```swift
+// MARK: Remote Notifications Lifecycle
+func application(_ application: UIApplication,
+    didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+    let tokenParts = deviceToken.map { data in String(format: "%02.2hhx", data) }
+    let token = tokenParts.joined()
+    print("Device Token: \(token)")
+
+    // Register the device token with Pinpoint as the endpoint for this user
+    pinpoint!.notificationManager.interceptDidRegisterForRemoteNotifications(withDeviceToken: deviceToken)
+}
+
+func application(_ application: UIApplication,
+    didFailToRegisterForRemoteNotificationsWithError error: Error) {
+    print("Failed to register: \(error)")
+}
+
+func application(_ application: UIApplication,
+                    didReceiveRemoteNotification userInfo: [AnyHashable: Any],
+                    fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+
+    if (application.applicationState == .active) {
+        let alert = UIAlertController(title: "Notification Received",
+                                        message: userInfo.description,
+                                        preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil))
+
+        UIApplication.shared.keyWindow?.rootViewController?.present(
+            alert, animated: true, completion:nil)
     }
-    ```
+
+    pinpoint!.notificationManager.interceptDidReceiveRemoteNotification(
+        userInfo, fetchCompletionHandler: completionHandler)
+}
+```
+
+3. Build and run the app. Allow notifications when prompted, and you should see the device token printed out. 
+
+4. Enable verbose logging for AWSPinpoint SDK, re-run the app, and you should be able to find the `endpointId` that is registered with AWSPinpoint. 
+```
+AWSDDLog.sharedInstance.logLevel = .verbose
+AWSDDLog.add(AWSDDTTYLogger.sharedInstance)
```

Comment:
two things here: 
1. where do I put these lines if I am new and have never done this before? pretty sure they go in didFinishLaunchingWithOptions but if you could be more specific with where to put these, that would be graet.
2. it says I can find `endpointId` but when I follow these instructions here is what I see in the log:
```
2020-01-24 12:45:37:071 PinpointSampleApp[40016:14331493] Response body:
{"RequestID":"4d5f5e4e-bcf6-463f-94bb-d16d8b70630b","Message":"Accepted"}
2020-01-24 12:45:37:072 PinpointSampleApp[40016:14331493] Endpoint Updated Successfully! <AWSPinpointTargetingUpdateEndpointResponse: 0x2825e8010> {
    messageBody = "<AWSPinpointTargetingMessageBody: 0x2827f03e0> {\n    message = Accepted;\n    requestID = \"4d5f5e4e-bcf6-463f-94bb-d16d8b70630b\";\n}";
}
```

Reply:
I don't see `endpointId` only requestId and that my endpoint was updated successfully

Reply:
thanks, for the first point, i added some context above the code snippet.

for the `endpointId`, i will have to look into this a bit more since endpointId was printed out with verbose logging, but i'm trying to avoid having the customer add that at this point of the docs

Reply:
Actually i do say to enable verbose logging for the endpointId. the customer can actually just use the device token to send push notification instead of the endpointId. I'm thinking of keeping the verbose logging step as optional and guiding the customer to use device token since that is printed out on successfully APNS registeration before pinpoint endpoint registrations